### PR TITLE
Fix CI and update dependencies

### PR DESCRIPTION
**Pull request by kibertoad**
Scope of this PR unfortunately grew quite a bit beyond quick "in and out" with dep upgrade that I hoped for; apparently CI was quite broken, so I had to fix CI scripts in order to prove that my upgrade doesn't break anything :D

### CODE REVIEW

**Comment by kibertoad** [comment]
https://github.com/gregjacobs/Autolinker.js/issues/278 -> latest autolinker could be used after this is resolved or clarified.

File: Makefile [hunk 1]
```diff
@@ -19,7 +19,7 @@ demo: lint
 	rm -rf demo/example.json
 
 lint:
-	eslint --reset ./bin ./lib ./support ./test
+	eslint ./bin ./lib ./support ./test
```
Comment (kibertoad) [inline Makefile]: Option is no longer there in newer ESLint, it's a new default behaviour.

File: test/utils.js [hunk 2]
```diff
@@ -23,11 +23,11 @@ function addTests(fPath, markdown, skip) {
         // Also skip tests if file name starts with "_"
         if (!skip && path.basename(fPath)[0] !== '_') {
           it('line ' + line, function () {
-            assert.strictEqual(html, markdown.render(md));
+            assert.strictEqual(markdown.render(md), html);
```
Comment (kibertoad) [inline test/utils.js]: expected and actual were mixed up

**Comment by kibertoad** [comment]
If/after this is merged, I can submit a separate PR that would resolve outstanding linting issues; didn't want to make this PR larger, though.

**Comment by TrySound** [comment]
Hi @kibertoad. Could you send a separate PR with autolinker update?

**Pull request closed without merging**